### PR TITLE
feat: Transactions chart page

### LINKED ISSUE #648: Transactions chart page
**Acceptance criteria:**
- A new page called "Charts" is added in the top menu
- A left navigation bar is visible in the page with a clickable item "Transactions"
- The right side of the page contains a chart with transactions data, same of https://aescan.io/transactions
- At the top of the chart:
  - Time interval controls are visible
  - A dropdown to select the resolution is visible with the following values: 1 Day, 1 Week, 1 Month
  - A dropdown with transactions type is visible, to filter the chart
- The default interval of the chart is the last year with a resolution of 1 day

### PR DESCRIPTION
**Pull request by janmichek**
## Description
Resolves #648

## Demo

https://github.com/aeternity/aescan/assets/15363559/efeb51c0-a160-4090-881f-53dae5769900

## Checklist:

- [x] I have read and followed the [Contributing Guide](../../CONTRIBUTING.md)  
- [x] add charts subpage
- [x] add transaction page chart

### CODE REVIEW

File: src/components/AppSelect.vue [hunk 1]
```diff
@@ -9,7 +9,10 @@
     :disabled="disabled"
     :searchable="searchable"
     :hide-selected="hideSelected"
-    :preselect-first="preselectFirst">
+    :preselect-first="preselectFirst"
+    :class="[
+      size ? `multiselect--${size}` : null,
```
Comment (janmichek) [inline src/components/AppSelect.vue:14]: add smaller version of select

File: src/components/LineChart.vue [hunk 2]
```diff
@@ -1,7 +1,14 @@
 <template>
-  <Line
-    :options="chartOptions"
-    :data="chartData"/>
+  <div class="line-chart">
+    <Line
+      v-if="hasChart"
+      :options="chartOptions"
+      :data="chartData"/>
+    <blank-state
```
Comment (janmichek) [inline src/components/LineChart.vue:7]: added loader when changing values of chart and blank state indicator for all charts

File: src/components/TheNavigation.vue [hunk 3]
```diff
@@ -71,6 +71,16 @@ const menuOptions = ref([{
       path: '/contract-verification',
     },
   ],
+},
+{
+  name: 'Charts',
```
Comment (janmichek) [inline src/components/TheNavigation.vue:76]: we can improve the navigation in following PRs, just a kickoff

File: src/pages/names/index.vue [hunk 4]
```diff
@@ -15,7 +15,7 @@
     </template>
   </page-header>
   <template v-if="!isLoading">
-    <names-chart-panel class="names_names-panel"/>
+    <names-chart-panel class="names__names-panel"/>
```
Comment (janmichek) [inline src/pages/names/index.vue:18]: typo

**Comment by janmichek** [comment]
From the initial requirements
`The default interval of the chart is the last year with a resolution of 1 day`
 I suggest (and implemented it)
`The default interval of the chart is the whole history with a resolution of 1 month`

**Review by Liubov-crypto — COMMENTED** [review]
LGTM.

I have one question: shouldn't transaction  be clickable and redirect user to transaction page? 


https://github.com/aeternity/aescan/assets/69896204/c0fa889e-e8b7-4701-98ba-8a693b17be20

**Comment by janmichek** [comment]
> LGTM.
> 
> I have one question: shouldn't Transactions be clickable and redirect user to transaction page?
> 2024-04-03.4.15.10.mov

No, not here. There will be more options for Blocks, Names, Smart Contracts ... in following implementations

**Comment by Liubov-crypto** [comment]
> > It seems to be broken for the All types Tx today in mainnet:
> > 2024-04-09.2.18.39.mov
> > errr
> > @janmichek could you check it please?
> 
> Ahaa, you are right. Fixed

Lgtm

**Review by Liubov-crypto — APPROVED** [review]

**Comment by janmichek** [comment]
> LGTM.
> 
> I would just make the transaction type dropdown slightly bigger to not make some types overflow, for example: image

fixed

**Comment by janmichek** [comment]
@michele-franchi one more question - 
Now we have 2 variants of transaction charts. WIth and without type filtering.
What we wanna do with the chart on /transactions subpage?
1. Leave as it is
2. display type filtering
3. bind select filter for table entries to chart as well

**Comment by michele-franchi** [comment]
@janmichek well, the dropdown to select the transaction should of course be visible only in the /charts page.
Having similar components is a different issue, couldn't it be the same component but where the dropdown is optional?

**Comment by janmichek** [comment]
> @janmichek well, the dropdown to select the transaction should of course be visible only in the /charts page. Having similar components is a different issue, couldn't it be the same component but where the dropdown can be optional with a prop?

True, can be abstracted some more. Fixed

**Pull request merged**